solidity_contracts/src/Kakarot/Coinbase.sol

```diff
@@ -1,23 +1,32 @@
 // SPDX-License-Identifier: MIT
 pragma solidity 0.8.27;
 
-import {DualVmToken} from "../CairoPrecompiles/DualVmToken.sol";
 import {Ownable2Step, Ownable} from "@openzeppelin-contracts/access/Ownable2Step.sol";
+import {CairoLib} from "kakarot-lib/CairoLib.sol";
 
 contract Coinbase is Ownable2Step {
-    /// @dev The EVM address of the DualVmToken for Kakarot ETH.
-    DualVmToken public immutable kakarotEth;
+    using CairoLib for uint256;
+
+    uint256 public immutable nativeTokenStarknetAddress;
 
     /// Constructor sets the owner of the contract
-    constructor(address _kakarotEth) Ownable(msg.sender) {
-        kakarotEth = DualVmToken(_kakarotEth);
+    constructor(uint256 _nativeTokenStarknetAddress) Ownable(msg.sender) {
+        nativeTokenStarknetAddress = _nativeTokenStarknetAddress;
     }
 
-    /// @notice Withdraws ETH from the contract to a Starknet address
-    /// @dev DualVmToken.balanceOf(this) is the same as address(this).balance
+    function receiveEther() external payable {}
```

Comment:
use regular fallback function

Reply:
> use regular fallback function

I don't want to spend more time trying to make our web3.py wrapper work with the `receive` special function. I can but at this point i'm out of ideas on how to make it work

Reply:
ok i made it work